pokemon-showdown

```diff
@@ -33,14 +33,16 @@ try {
 } catch (e) {}
 
 if (config && config.golang) {
-	if (!process.env.GOPATH) {
-		console.log('The GOPATH environment variable is not set! It is required in order to run the server using Go.');
-		process.exit(0);
-	}
-	if (!process.env.GOROOT) {
-		console.log('The GOROOT environment variable is not set! It is required in order to run the server using Go.');
-		process.exit(0);
-	}
+	// GOPATH and GOROOT are optional to a degree, but we need them in order
+	// to be able to handle Go dependencies. Since Go only cares about the
+	// first path in the list, so will we.
+	const GOPATH = child_process.execSync('go env GOPATH', {stdio: null, encoding: 'utf8'})
+		.trim().split(path.delimiter)[0];
+	process.env.GOPATH = GOPATH;
+
+	let GOROOT = child_process.execSync('go env GOROOT', {stdio: null, encoding: 'utf8'})
```

Comment:
[Gauging from this](https://github.com/golang/go/issues/17262) it looks like we'll still have to check if `$GOPATH` is set if the version of Go that users are using is below 1.8 and either only stop them from running PS with Go if it's not set or deny them completely. I don't like the idea of changing server admins' env vars implicitly